Frameworks/MQTTnet.NetCoreApp/MqttSslServerAdapter.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using MQTTnet.Core.Adapter;
+using MQTTnet.Core.Serializer;
+using MQTTnet.Core.Server;
+
+namespace MQTTnet
+{
+    public class MqttSslServerAdapter : IMqttServerAdapter, IDisposable
+    {
+        private CancellationTokenSource _cancellationTokenSource;
+        private Socket _socket;
+        private X509Certificate2 _x590Certificate2;
+
+        public event EventHandler<MqttClientConnectedEventArgs> ClientConnected;
+
+        public void Start(MqttServerOptions options)
+        {
+            if (_socket != null) throw new InvalidOperationException("Server is already started.");
+
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            _socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+            _socket.Bind(new IPEndPoint(IPAddress.Any, options.Port));
+            _socket.Listen(options.ConnectionBacklog);
+
+            Task.Run(async () => await AcceptConnectionsAsync(_cancellationTokenSource.Token), _cancellationTokenSource.Token);
+
+            _x590Certificate2 = new X509Certificate2(options.CertificatePath);
```

Comment:
It is a very low chance but the task should be started if everything is assinged so that no null-ref is possible if the cert is used but not set.